File: fix_imports.py
import os
import re

directory = r'i:\KMTI_MG_APP_DEVELOPMENT\kmti-icad-hub\frontend\src\components\ICAD\Manual'

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts') or file.endswith('.css'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Replace ../../ with ../../../../
            new_content = content.replace('../../', '../../../../')
            
            # Replace from "../ or from '../ with from "../../../ or from '../../../
            new_content = re.sub(r'from "(?!\.\.)\.\./([A-Za-z])', r'from "../../../\1', new_content)
            new_content = re.sub(r"from '(?!\.\.)\.\./([A-Za-z])", r"from '../../../\1", new_content)
            
            if new_content != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)

File: fix_imports.js
const fs = require('fs');
const path = require('path');

function walkDir(dir) {
    fs.readdirSync(dir).forEach(file => {
        let fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts') || fullPath.endsWith('.css')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let original = content;
            
            content = content.replace(/from\s+["']\.\.\/\.\.\//g, 'from "../../../../');
            content = content.replace(/import\s+["']\.\.\/\.\.\//g, 'import "../../../../');
            content = content.replace(/url\(['"]?\.\.\/\.\.\//g, 'url(\'../../../../');

            content = content.replace(/from\s+["']\.\.\/(?!\.\.\/)/g, 'from "../../../');
            content = content.replace(/import\s+["']\.\.\/(?!\.\.\/)/g, 'import "../../../');
            
            if (content !== original) {
                fs.writeFileSync(fullPath, content, 'utf8');
            }
        }
    });
}

walkDir('i:/KMTI_MG_APP_DEVELOPMENT/kmti-icad-hub/frontend/src/components/ICAD/Manual');
